src/main/java/com/zaxxer/nuprocess/linux/ProcessEpoll.java

```diff
@@ -38,23 +36,12 @@
  */
 class ProcessEpoll extends BaseEventProcessor<LinuxProcess>
 {
-   private static final int EVENT_POOL_SIZE = 64;
-   private static final BlockingQueue<EpollEvent> eventPool;
+   private final int epoll;
+   private final EpollEvent triggeredEvent;
+   private final List<LinuxProcess> deadPool;
 
-   private int epoll;
-   private EpollEvent triggeredEvent;
-   private List<LinuxProcess> deadPool;
    private LinuxProcess process;
 
-   static
-   {
-      eventPool = new ArrayBlockingQueue<>(EVENT_POOL_SIZE);
```

Comment:
Bitbucket Server under load can be running hundreds of processes concurrently and, since it generally uses `NuProcess.run` instead of `NuProcess.start`, it can be pumping those processes on hundreds of threads. Under those conditions, it's easy for the system to run out of events in the queue and block waiting for one to be returned. If a blocked thread got interrupted (e.g. because a request was canceled), the exception thrown would end up causing NuProcess to abandon the process, resulting in a zombie.

This static queue was added to combat memory churn (see 9df9b4ccdc055662c0406890fb1e419211ca694d), but in practice there's no observable difference in churn (at least per the jmh microbenchmark that was used to analyze the change above when it was made) between having a reusable `EpollEvent` on each `LinuxProcess` and using this static queue. Given that, this approach of an event per process is simpler and avoids any contested resources.